content/en/docs/ibm/deploy/authentication.md

```diff
@@ -145,3 +145,130 @@ The secret in the `default` namespace is automatically renewed by IBM Cloud
+
+2. Since the certificates provided by IBM Cloud only allow for a single level of domain name added
+to the base domain, the domain template for Knative needs to be adjusted so that the certificates
+will be valid for the InferenceService routes.
+
+    ```shell
+    kubectl edit configmap config-network -n knative-serving
+    ```
+
+    This will open your default text editor, and you will again see something like:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      _example: |
+        ################################
+        #                              #
+        #    EXAMPLE CONFIGURATION     #
+        #                              #
+        ################################
+        # ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Add a line above the `_example` key with the `domainTemplate` key like the following:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      domainTemplate: "{{.Name}}-{{.Namespace}}.{{.Domain}}"
+      _example: |
+       ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Save and exit. The default template uses a two-level subdomain (i.e. `{{.Name}}.{{.Namespace}}.{{.Domain}}`),
+    so this just adjusts it to use one.
+
+
+3. Adjust `kubeflow-gateway` one more time, adding a wildcard host in the HTTPS `hosts` section.
+
+    ```shell
+    kubectl edit gateway kubeflow-gateway -n kubeflow
+    ```
+
+    This will open your default text editor, but you can also optionally edit `kubeflow-gateway.yaml` file
+    you created previously. Here, just add another entry to the HTTPS `hosts` list containing your hostname prepended
+    with a `*.` so that the Knative subdomains are properly routed.
+
+    ```YAML
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: kubeflow-gateway
+      namespace: kubeflow
+    spec:
+      selector:
+        istio: ingressgateway
+      servers:
+      - hosts:
+        - '<hostname>'
+        - '*.<hostname>'
+        port:
+          name: https
+          number: 443
+          protocol: HTTPS
+        tls:
+          mode: SIMPLE
+          privateKey: /etc/istio/ingressgateway-certs/tls.key
+          serverCertificate: /etc/istio/ingressgateway-certs/tls.crt
+    ```
+
+    Save and exit.
+
+After these adjustments, InferenceServices should now be reachable via HTTPS. However, if you are using IBM Cloud AppID for
+authentication and you want to predict externally, you will need to generate a token as
+[described here](https://cloud.ibm.com/docs/appid?topic=appid-custom-auth#generating-jwts). This token can be used as a header
+(e.g. `"Authorization: Bearer eyJh..."`) in your request.
+
+Alternatively, for testing, you can use the `authservice_session` cookie for the Kubeflow dashboard site from the browser. Once
+the content of the cookie is retrieved from the browser, it can be added as a header in your request
+(e.g. `"Cookie: authservice_session=MTYwNz..."`).
```

Comment:
If the request is sending over CURL, it should use the header `X-Auth-Token: <token wtihout authservice_session from cookie> `. but cookie works, I won't block merge if using cookie header.

Reply:
Hmm, I can't seem to get that header to work.

This works:

```
curl -v -H "Cookie: authservice_session=MTYwODIzNDQyOHxOd3dBTkZoSk5VRTNNek5QUVRKR1NFTlNWRU5ITWpKRVJ6TTJWamRFVTA1RVNrbFhSelZOU0ZkUFZFeEtRMWRRVFRaS1RVeFNObEU9fPBUEE3vaC4PIxxHRwuphwFWf0DdbIDy-VA9_DQ2yuq5" https://sklearn-iris-bar-p-test.kfp-dev-442dbba0442be6c8c50f31ed96b00601-0001.sjc03.containers.appdomain.cloud/v1/models/sklearn-iris-bar:predict -d '{"instances": [[6.8,  2.8,  4.8,  1.4],[6.0,  3.4,  4.5,  1.6]]}'
```
But not this:
```
curl -v -H "X-Auth-Token: MTYwODIzNDQyOHxOd3dBTkZoSk5VRTNNek5QUVRKR1NFTlNWRU5ITWpKRVJ6TTJWamRFVTA1RVNrbFhSelZOU0ZkUFZFeEtRMWRRVFRaS1RVeFNObEU9fPBUEE3vaC4PIxxHRwuphwFWf0DdbIDy-VA9_DQ2yuq5" https://sklearn-iris-bar-p-test.kfp-dev-442dbba0442be6c8c50f31ed96b00601-0001.sjc03.containers.appdomain.cloud/v1/models/sklearn-iris-bar:predict -d '{"instances": [[6.8,  2.8,  4.8,  1.4],[6.0,  3.4,  4.5,  1.6]]}' 
```

Reply:
I saw in Kubeflow v1.2, it still uses `cookie` header and `X-Auth-Token` header in the enovyfilter that forwards request to auth service: https://github.com/kubeflow/manifests/blob/b64b76b4ef3a984115f01686812588e81f46a02b/istio/oidc-authservice/base/envoy-filter.yaml#L19-L20

So `X-Auth-Token` should work, but anyway, won't block merge here.